sources/common/infoschema.go

```diff
@@ -133,6 +139,141 @@ func ProcessData(conv *internal.Conv, infoSchema InfoSchema) {
 	}
 }
 
+func ProcessDataWithDataproc(conv *internal.Conv, infoSchema InfoSchema, dataprocConfig map[string]string) error {
+
+	orderTableNamesByID := ddl.GetSortedTableIdsBySpName(conv.SpSchema)
```

Comment:
This would return a list of tables taking interleaving into account, but since this method only gives out a `List`, it won't be possible to parallelise Dataproc template execution.

We should instead do the following - 

1. Implement a method which would return a `List<List>`, each `List` would contain 1 or more elements, ordered from parent to children.
2. Create a fixed size worker pool to process these lists. Each worker takes a `List` as an input.
3. Each worker serially processes the List and orchestrates Dataproc template.

There are some code refs in HB for worker pools -

https://github.com/cloudspannerecosystem/harbourbridge/blob/master/conversion/conversion.go#L337

For the method implementation - We need to represent the DDL has an adjacency list with root nodes being parent tables.

Reply:
As discussed, We'll support only serial execution (respecting interleaving) as part of this PR, and add support for parallel execution in upcoming PR